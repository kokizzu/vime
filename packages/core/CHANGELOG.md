# Change Log

All notable changes to this project will be documented in this file.

## [1.0.2](https://github.com/vime-js/vime/compare/@vime/core@1.0.1...@vime/core@1.0.2) (2020-08-13)


### Bug Fixes

* **core:** point unpkg and jsdelivr to correct file ([8360ad6](https://github.com/vime-js/vime/commit/8360ad62197ab35a3f51986f83e9412c7fd11a3b))

## [1.0.1](https://github.com/vime-js/vime/compare/@vime/core@1.0.0...@vime/core@1.0.1) (2020-08-13)


### Bug Fixes

* **core:** icons not loading ([2b48c85](https://github.com/vime-js/vime/commit/2b48c8547b0b7454b48a2c2707e82d74837ae9cb))

# 1.0.0 (2020-08-12)


### Bug Fixes

* **core:** lazy loading broken ([792520a](https://github.com/vime-js/vime/commit/792520ac17dcd8d70179ac4b29d91d4ac5a127eb))
* **core:** prevent posters loading twice with custom UI ([5aefb06](https://github.com/vime-js/vime/commit/5aefb06f3ad7e80204ce4966f42d5372a3517a70))
* **core/ui:** slider keydown events propagating ([94fafe1](https://github.com/vime-js/vime/commit/94fafe1f485e841febea3dd8a065588bf874622a))
* **core/ui/captions:** emitting events after disconnecting ([5f0abac](https://github.com/vime-js/vime/commit/5f0abac07f6b497607968265435e76a248ba3edf))
* **core/ui/captions:** toggling captions on/off broken ([090d776](https://github.com/vime-js/vime/commit/090d7762401750d91f9a1cc12a320cbfa3652eb6))


### Features

* **core:** add lazy loading via lozad ([3bde06c](https://github.com/vime-js/vime/commit/3bde06c5928895195f8e934c6808eca964737d07))
* **core:** add skeleton loading animation ([ff7860d](https://github.com/vime-js/vime/commit/ff7860ddfe74f029c3d365870a31d6d3d72f2748))
* **core:** dailymotion + hls + dash providers ([c520c30](https://github.com/vime-js/vime/commit/c520c302e5db91f20c9944915a43102ff1305e51))
* **core:** improve test harness + vimeo provider ([dd153ca](https://github.com/vime-js/vime/commit/dd153ca96026a3ef80c89fa1dcc560e36dcc7c1d))
* **core:** vime-embed component ([b45be6f](https://github.com/vime-js/vime/commit/b45be6f642e286b7da568b4450e8268334961019))
* **core:** vime-player component ([c86bee4](https://github.com/vime-js/vime/commit/c86bee4f709fd51d6457805df6c6c19efa0abc89))
* **core:** youtube provider + providers test harness ([8f93a68](https://github.com/vime-js/vime/commit/8f93a68edd16d4396c708e84e8e5bc1d8da69c41))
* **core/ui:** add click-to-play component ([fcce4f1](https://github.com/vime-js/vime/commit/fcce4f11b32f4603f96677dd10803233c344e8d8))
* **core/ui:** add light audio player theme ([93d4206](https://github.com/vime-js/vime/commit/93d42062ee2515ac37a10244961767714cfa87f9))
* **core/ui:** captions component ([c810f42](https://github.com/vime-js/vime/commit/c810f42d00cc81aa3cf35bf81baf2846dcbba4bb))
* **core/ui:** controls components ([5dc4e26](https://github.com/vime-js/vime/commit/5dc4e261e23df6993694abfa776af8637fb7b9a6))
* **core/ui:** create settings components ([3a1fd82](https://github.com/vime-js/vime/commit/3a1fd82a232b67f27759dc94e1e76fea8228f97e))
* **core/ui:** icons + icon component ([d13df8e](https://github.com/vime-js/vime/commit/d13df8ef1b265ec995b8a37f023162f7a978da37))
* **core/ui:** poster + spinner components ([cbdd69f](https://github.com/vime-js/vime/commit/cbdd69ff9f3474639f2a781809859c3b94beb7ff))
* **core/ui:** scrim component ([df47efe](https://github.com/vime-js/vime/commit/df47efe31b3088157896120b9f7ff2d78bdd5c99))
* **core/ui/controls:** add option to scale control size up/down ([3635b1a](https://github.com/vime-js/vime/commit/3635b1ae0d2d2d0163086a01d4e1f6fb2468a561))
* **core/ui/poster:** add poster events ([9d36f47](https://github.com/vime-js/vime/commit/9d36f47a55f6459e8ffc24018cf2fb5c2654fcde))
* **core/ui/scrim:** add events ([1180240](https://github.com/vime-js/vime/commit/1180240c8e93cce96009c3c2098064806dfa4c37))
* **core/ui/scrim:** add gradient option ([adf6bcb](https://github.com/vime-js/vime/commit/adf6bcbdb2d8520552bb6c93b5bc1f30ab95ce84))
* **core/ui/settings:** add audio/mobile styles to settings ([c937312](https://github.com/vime-js/vime/commit/c937312b344e2a5b3bb10315367f98c6b0eb81a0))
* **core/ui/spinner:** add events ([13dfd14](https://github.com/vime-js/vime/commit/13dfd1445d72357a1ed315cd3a25e955be19a772))
