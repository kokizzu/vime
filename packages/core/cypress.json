{
  "baseUrl": "http://localhost:3333/src/components",
  "watchForFileChanges": false,
  "integrationFolder": "cypress/tests",
  "testFiles": "**/*.e2e.ts",
  "fixturesFolder": false,
  "supportsFolder": "cypress/support/index.ts",
  "ignoreTestFiles": [
    "**/__snapshots__/*",
    "**/__image_snapshots__/*"
  ],
  "env": {
    "cypress-plugin-snapshots": {
      "imageConfig": {
        "threshold": 0.05
      }
    }
  }
}
