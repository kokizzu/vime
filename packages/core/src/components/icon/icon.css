 vime-icon > svg {
  display: block;
  width: 18px;
  height: 18px;
  fill: currentColor;
  pointer-events: none;
}
