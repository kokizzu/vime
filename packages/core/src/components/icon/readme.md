# vime-icon



<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                                                                               | Type                  | Default     |
| --------- | --------- | ----------------------------------------------------------------------------------------- | --------------------- | ----------- |
| `color`   | `color`   | The color (fill) of the icon.                                                             | `string`              | `'#fff'`    |
| `icon`    | `icon`    | The icon SVG identifier. It's expected that this points to an SVG inside a loaded sprite. | `string \| undefined` | `undefined` |
| `opacity` | `opacity` | The amount of transparency to add to the icon.                                            | `number`              | `1`         |
| `scale`   | `scale`   | The amount to scale the size of the icon (respecting aspect ratio) up or down by.         | `number`              | `1`         |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
