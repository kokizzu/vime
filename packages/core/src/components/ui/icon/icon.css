 vime-icon > svg {
  /**
   * @prop --icon-width: The width of the icon.
   * @prop --icon-height: The height of the icon.
   */

  display: block;
  width: var(--icon-width, 18px);
  height: var(--icon-height, 18px);
  fill: currentColor;
  pointer-events: none;
}
