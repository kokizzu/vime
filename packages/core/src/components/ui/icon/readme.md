# vime-icon

Renders and displays an SVG icon. The SVG markup can be passed in directly or it can be loaded via 
a URL. It's preferrable to load a sprite via `vime-icons` and reference the icon using the `href` 
property.

## Example

```html
<!-- Markup -->
<vime-icon>
  <rect width="300" height="100" />
</vime-icon>

<!-- URL -->
<vime-icon href="#vime-play"></vime-icon>
```

<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                                                                       | Type                  | Default     |
| --------- | --------- | --------------------------------------------------------------------------------- | --------------------- | ----------- |
| `color`   | `color`   | The color (fill) of the icon.                                                     | `string`              | `'#fff'`    |
| `href`    | `href`    | The URL to an SVG element or fragment to load.                                    | `string \| undefined` | `undefined` |
| `opacity` | `opacity` | The amount of transparency to add to the icon.                                    | `number`              | `1`         |
| `scale`   | `scale`   | The amount to scale the size of the icon (respecting aspect ratio) up or down by. | `number`              | `1`         |


## Slots

| Slot | Description                            |
| ---- | -------------------------------------- |
|      | SVG markup to be drawn by the browser. |


## CSS Custom Properties

| Name            | Description             |
| --------------- | ----------------------- |
| `--icon-height` | The height of the icon. |
| `--icon-width`  | The width of the icon.  |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
