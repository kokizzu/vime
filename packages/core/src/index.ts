import { PlayerProp, PlayerProps } from './components/core/player/PlayerProp';
import { PlayerDispatcher, createPlayerDispatcher } from './components/core/player/PlayerDispatcher';
import { withPlayerContext, usePlayerContext } from './components/core/player/PlayerContext';
import { PlayerEvent } from './components/core/player/PlayerEvent';
import { findRootPlayer } from './components/core/player/utils';

export * from './components';

export {
  PlayerProp,
  PlayerProps,
  PlayerEvent,
  findRootPlayer,
  withPlayerContext,
  usePlayerContext,
  PlayerDispatcher,
  createPlayerDispatcher,
};
