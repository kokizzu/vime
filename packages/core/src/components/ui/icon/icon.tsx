import { Component, h, Prop } from '@stencil/core';
import { isString } from '../../../utils/unit';

/**
 * @slot - Used to pass in SVG markup to be drawn by the browser.
 */
@Component({
  tag: 'vime-icon',
  styleUrl: 'icon.css',
})
export class Icon {
  /**
   * The URL to an SVG element or fragment to load.
   */
  @Prop() href?: string;

  /**
   * The color (fill) of the icon.
   */
  @Prop() color?: string;

  /**
   * The amount to scale the size of the icon (respecting aspect ratio) up or down by.
   */
  @Prop() scale = 1;

  /**
   * The amount of transparency to add to the icon.
   */
  @Prop() opacity = 1;

  render() {
    return (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        role="presentation"
        focusable="false"
        style={{
          color: this.color,
          transform: `scale(${this.scale})`,
          opacity: `${this.opacity}`,
        }}
      >
        {isString(this.href) ? <use href={this.href as string} /> : <slot />}
      </svg>
    );
  }
}
