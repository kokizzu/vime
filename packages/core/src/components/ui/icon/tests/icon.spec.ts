import { newSpecPage } from '@stencil/core/testing';
import { Icon } from '../icon';

it('should be structurally sound', async () => {
  const page = await newSpecPage({
    components: [Icon],
    html: '<vime-icon icon="#my-icon"></vime-icon>',
  });

  expect(page.root).toMatchInlineSnapshot(`
    <vime-icon icon="#my-icon">
      <!---->
      <svg focusable="false" role="presentation" xmlns="http://www.w3.org/2000/svg" style="color: #fff; transform: scale(1); opacity: 1;">
        <use href="#my-icon"></use>
      </svg>
    </vime-icon>
  `);
});

it('should accept a slotted inline-svg', async () => {
  const page = await newSpecPage({
    components: [Icon],
    html: `
      <vime-icon>
        <rect width="300" height="100" />
      </vime-icon>
    `,
  });

  expect(page.root).toMatchInlineSnapshot(`
    <vime-icon>
      <!---->
      <svg focusable="false" role="presentation" xmlns="http://www.w3.org/2000/svg" style="color: #fff; transform: scale(1); opacity: 1;">
        <rect height="100" width="300"></rect>
      </svg>
    </vime-icon>
  `);
});
