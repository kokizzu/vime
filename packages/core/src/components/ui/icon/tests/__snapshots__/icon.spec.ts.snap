// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept svg markup 1`] = `
<vime-icon>
  <!---->
  <svg focusable="false" role="presentation" xmlns="http://www.w3.org/2000/svg" style="transform: scale(1); opacity: 1;">
    <rect height="100" width="300"></rect>
  </svg>
</vime-icon>
`;

exports[`should be structurally sound 1`] = `
<vime-icon href="#my-icon">
  <!---->
  <svg focusable="false" role="presentation" xmlns="http://www.w3.org/2000/svg" style="transform: scale(1); opacity: 1;">
    <use href="#my-icon"></use>
  </svg>
</vime-icon>
`;
